tapdb/multiverse.go

```diff
@@ -133,11 +541,133 @@ func (b *MultiverseStore) RootNode(ctx context.Context,
 	return rootNode, nil
 }
 
+// UniverseRootNode returns the Universe root node for the given asset ID.
+func (b *MultiverseStore) UniverseRootNode(ctx context.Context,
+	id universe.Identifier) (universe.Root, error) {
+
+	// First, we'll check the root node cache to see if we already have it.
+	rootNode := b.rootNodeCache.fetchRoot(id)
+	if rootNode != nil {
+		return *rootNode, nil
+	}
+
+	b.rootNodeCache.Lock()
```

Comment:
Was going to suggest something like this in the earlier commits; I think some of the other caches aren't using a RWMutex though?

Can address in follow-up PR.

Reply:
The mutex here is in place so that when the cache needs to be populated, only one goroutine does, and not like 10k.